### PR TITLE
Improve out-of-memory error message

### LINKED ISSUE #2241: "total" memory in OOM error message is confusing
What "total" memory mentioned in the error message means is ambiguous.

```
cupy.cuda.memory.OutOfMemoryError: out of memory to allocate 200000000000 bytes (total 200000004096 bytes)
```

I'd imagine it would be either the total available memory in the system, or the total size of memory allocated so far.
Actually it's neither: the total size of memory that *would be* allocated if the allocation suceeded.

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #2241

* Show the total memory successfully allocated so far, and explicitly state what it is.
* Put commas
* Wording improvement

### Before
```
cupy.cuda.memory.OutOfMemoryError: out of memory to allocate 200000000000 bytes (total 200000004096 bytes)
```

### This PR
```
cupy.cuda.memory.OutOfMemoryError: Out of memory allocating 200,000,000,000 bytes (allocated so far: 4,096 bytes).
```

### CODE REVIEW

File: cupy/cuda/memory.pyx [hunk 1]
```diff
@@ -34,14 +34,24 @@ def _exit():
...
+                '(allocated so far: {:,} bytes).'.format(size, total))
         else:
-            msg = ('out of memory to allocate {} bytes '
-                   '(total {} bytes, limit set to {} bytes)'.format(
+            msg = (
+                'Out of memory allocating {:,} bytes '
+                '(allocated so far: {:,} bytes, '
+                'limit set to: {:,} bytes).'.format(
                        size, total, limit))
         super(OutOfMemoryError, self).__init__(msg)
```
Comment (emcastillo) [inline cupy/cuda/memory.pyx]: Flake8 is complaining on this line

```
./cupy/cuda/memory.pyx:55:24:  E126 continuation line over-indented for hanging indent
The command "flake8 --config=.flake8.cython" exited with 1.
```

collapsing format.(size, total, limit) in one line does not complain anymore

Comment (niboshi) [reply cupy/cuda/memory.pyx]: Thank you for pointing out.
Fixed (force-pushed) as suggested.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins error is not related to this so I approve the merge

**Pull request merged**